src/go/rpk/pkg/cli/cmd/topic/consume.go

```diff
@@ -524,7 +540,12 @@ func (c *consumer) parseTimeOffset(
 	}
 	offset = offset[1:] // strip start:end delimiting colon
 
-	length, endAt, end, err := parseConsumeTimestamp(offset)
+	if fromT0 {
+		length, endAt, end, _, err = parseConsumeTimestamp(offset, startAt)
+	} else {
+		length, endAt, end, _, err = parseConsumeTimestamp(offset)
+	}
```

Comment:
If following my comments above, this could be:

```go
        if fromTimestamp {
                length, endAt, end, _, err = parseConsumeTimestamp(offset, startAt)
        } else {
                length, endAt, end, _, err = parseConsumeTimestamp(offset, time.Now())
        }
```